### ISSUE TITLE
Is there a way to drop only specific types of console logging ?

### ISSUE THREAD

**Issue by ishandutta2007** [evanw/esbuild#3656]
I have been using configuration to remove console logging with arguments like this,
```
drop: ['console', 'debugger'],
```
Now I am wondering if it is possible to only remove specific type of log like for example in my case I want to remove `console.log`,`console.info`,but I want to keep all the `console.error`

**Comment by evanw** [comment]
No, there isn't currently. This feature was introduced because of a similar option in [Terser](https://terser.org/). It looks like Terser has the ability to only drop specific types of logging, so it seems reasonable to add that to esbuild as well.

**Comment by th0rgall** [comment]
I'd also be interested in an esbuild implementation of this feature, if possible in such a way that it could also target arbitrary functions/expressions as well, like logging wrappers (`logger.info()` ...)

I tried to use this solution using `pure` https://github.com/vitejs/vite/discussions/7920#discussioncomment-4813594 in my SvelteKit build to only remove `logger.debug` statements, but it was not working for me, probably because `pure` behavior isn't exactly the same as `drop`.

As a workaround, I prepended every `logger.debug` statement with a `DEV: ` label and then used the `dropLabels` esbuild config https://esbuild.github.io/api/#drop-labels to drop those statements. It's more verbose but it seems to work reliably.
